misk-grpc-tests/src/test/kotlin/misk/grpc/MiskClientMiskServerTest.kt

```diff
@@ -105,4 +110,55 @@ class MiskClientMiskServerTest {
       sendChannel.close()
     }
   }
+
+  @Test
+  fun serverFailureGeneric() {
+    val point = Point(
+      latitude = -1,
+      longitude = 500
+    )
+
+    runBlocking {
+      val e = assertFailsWith<GrpcException> {
+        routeGuide.GetFeature().execute(point)
+      }
+      assertThat(e.grpcMessage).isEqualTo("unexpected latitude error!")
+      assertThat(e.grpcStatus).isEqualTo(GrpcStatus.UNKNOWN)
+
+      assertResponseCount(200, 0)
+      assertResponseCount(500, 1)
```

Comment:
This test is broken; will approve when it passes.

Also, nit: add code comment about why these are the expected metrics

Reply:
Can you be more specific about what you mean by "broken" here? The assertion is about what's counted in metrics, which we very much do want to be what's asserted here. See the discussion in https://github.com/cashapp/misk/pull/1983#discussion_r655746280

Reply:
Ah I saw a test failure here, but it looks like it's working now.